### PR TITLE
Wa 89

### PR DESCRIPTION
**Pull request by Jeva92**

### CODE REVIEW

File: src/http/login-http.js [hunk 1]
```diff
@@ -54,13 +54,25 @@ const login = createJsonRoute(function(req, res) {
...
+              .then(row => {
+                if (_.isEmpty(row)) {
+                  return throwStatus(500, 'User linking failed')
+                }})
+                return {admin: true};
+          } else {
+            return knex('role').select('admin').where('email', email).first()
+              .then(admin => {
+                return { admin: admin.admin, token: tokenForUser(id.id) };
+          })}
```
Comment (kaupunki-apina) [inline src/http/login-http.js]: Try and have database interactions in `core/` folder.

File: src/util/passport.js [hunk 2]
```diff
@@ -39,28 +39,42 @@ const localLogin = new LocalStrategy(localOptions, (email, password, done) => {
...
-      const id = payload.sub;
-      return done(null, id)
-    }
-  });
+const jwtLogin = new Strategy(jwtOptions, (req, payload, done) => {
+  const uuid = req.headers['x-user-uuid'];
+  return knex('users').select('id').where('uuid', uuid)
+    .then(rows => {
+      if (_.isEmpty(rows)) {
+        return knex('role').select('email').where('id', payload.sub)
```
Comment (kaupunki-apina) [inline src/util/passport.js:51]: There's multiple database queries in a row. I'd consider using `transactions`.

File: src/http/login-http.js (hunk 1 above)
Comment (Jeva92) [reply src/http/login-http.js]: Yes, the plan is to move all DB stuff into core folder, but as of now my main consern has been to actually get things to work ":D".

File: src/util/passport.js (hunk 2 above)
Comment (Jeva92) [reply src/util/passport.js:51]: Will look into this.

**Pull request merged**